src/Tracing/DynamicSamplingContext.php

```diff
@@ -0,0 +1,205 @@
+     * @return array<string, string>
+     */
+    public function getEntries(): array
+    {
+        return $this->entries;
+    }
+
+    /**
+     * Parse the baggage header.
+     *
+     * @param string $header the baggage header contents
+     */
+    public static function fromHeader(string $header): self
+    {
+        $dsc = new self();
+
+        foreach (explode(',', $header) as $listMember) {
+            if (empty(trim($listMember))) {
+                continue;
+            }
+
+            $keyValueAndProperties = explode(';', $listMember, 2);
+
+            $keyValue = trim($keyValueAndProperties[0]);
+
+            if (empty($keyValue) || !str_contains($keyValue, '=')) {
+                continue;
+            }
+
+            [$key, $value] = explode('=', $keyValue, 2);
+
+            if (0 === strpos($key, self::SENTRY_ENTRY_PREFIX)) {
+                $dsc->set(rawurldecode(str_replace(self::SENTRY_ENTRY_PREFIX, '', $key)), rawurldecode($value));
+            }
+        }
+
+        // Once we receive a baggage header with Sentry entries from an upstream SDK,
+        // we freeze the contents so it cannot be mutated anymore by this SDK.
+        // It should only be propagated to the next downstream SDK or the Sentry server itself.
+        $dsc->isFrozen = $dsc->hasEntries();
+
+        return $dsc;
+    }
+
+    /**
+     * Create a dsc object.
+     *
+     * @see https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#baggage-header
+     */
+    public static function fromTransaction(Transaction $transaction, HubInterface $hub): self
+    {
+        $dsc = new self();
+
+        $dsc->set('trace_id', (string) $transaction->getTraceId());
+        $dsc->set('sample_rate', (string) $transaction->getMetaData()->getSamplingRate());
+        $dsc->set('transaction', $transaction->getName());
+
+        $client = $hub->getClient();
+
+        if (null !== $client) {
+            $options = $client->getOptions();
+
+            if (null !== $options) {
+                if (null !== $options->getDsn() && null !== $options->getDsn()->getPublicKey()) {
+                    $dsc->set('public_key', $options->getDsn()->getPublicKey());
+                }
+                if (null !== $options->getRelease()) {
+                    $dsc->set('release', $options->getRelease());
+                }
+                if (null !== $options->getEnvironment()) {
+                    $dsc->set('environment', $options->getEnvironment());
+                }
+            }
+        }
+
+        $hub->configureScope(function (Scope $scope) use ($dsc) {
+            if (null !== $scope->getUser() && null !== $scope->getUser()->getSegment()) {
+                $dsc->set('user_segment', $scope->getUser()->getSegment());
+            }
+        });
+
+        $dsc->freeze();
+
+        return $dsc;
+    }
+
+    /**
+     * Serialize the dsc as a string.
+     */
+    public function __toString(): string
+    {
+        $string = '';
```

Comment:
Instead of building the string ourselves, wouldn't be easier to collect the parts into an array and then simply call `implode`? This would mean not bothering about the handling of the commas